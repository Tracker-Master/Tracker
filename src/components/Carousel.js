import React from 'react';

import { StyledCarousel, Carousell } from 'Styles/components/CarouselStyles';

export const Carousel = ({ children }) => {
  return (
    <StyledCarousel>
      <Carousell
        autoPlaySpeed={3000}
        enableAutoPlay={true}
        focusOnSelect={false}
        showArrows={false}
      >
        <img
          src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_one.svg"
          alt="carousel-slide-1"
        />
        <img
          src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_two.svg"
          alt="carousel-slide-2"
        />
        <img
          src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_three.svg"
          alt="carousel-slide-3"
        />
      </Carousell>
      <Carousell
        autoPlaySpeed={3000}
        enableAutoPlay={true}
        focusOnSelect={false}
        showArrows={false}
      >
        <p>
          Sólo necesitas introducir el ID de tu tablero y nosotros haremos el
          resto
        </p>
        <p>
          Sabrás el número de miembros con los que cuenta tu tablero, y el
          nombre de cada uno de ellos
        </p>
        <p>
          Sabrás de una manera rápida el número total de tarjetas que tienes en
          cada lista
        </p>
      </Carousell>
    </StyledCarousel>
  );
};
