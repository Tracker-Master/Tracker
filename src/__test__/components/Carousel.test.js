import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Carousel from '../../components/Carousel';

describe('<Carousel />', () => {
  
  const component = shallow(<Carousel />);

  test('Carousel rendering', () => {
    expect(component.length).toBe(1);
  });

  test(`Carousel's snapshot`, () => {
    const wrapper = create(<Carousel />);
    expect(wrapper).toMatchSnapshot();
  });


});