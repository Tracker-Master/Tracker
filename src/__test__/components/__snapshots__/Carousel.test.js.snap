// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> Carousel's snapshot 1`] = `
.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  overflow: hidden;
  position: relative;
  width: 100%;
  margin: 0 10px;
}

.c5 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  direction: ltr;
}

.c8 {
  box-sizing: border-box;
  padding: 0;
  -webkit-transition: all 250ms ease;
  transition: all 250ms ease;
  border: none;
  margin: 5px;
  background-color: rgba(103,58,183,.5);
  font-size: 1.3em;
  content: "";
  height: 10px;
  width: 10px;
  box-shadow: 0 0 1px 3px rgba(103,58,183,1);
  border-radius: 50%;
  outline: none;
}

.c8:hover,
.c8:focus {
  cursor: pointer;
  box-shadow: 0 0 1px 3px rgba(103,58,183,1);
}

.c9 {
  box-sizing: border-box;
  padding: 0;
  -webkit-transition: all 250ms ease;
  transition: all 250ms ease;
  border: none;
  margin: 5px;
  background-color: transparent;
  font-size: 1.3em;
  content: "";
  height: 10px;
  width: 10px;
  box-shadow: 0 0 1px 2px rgba(0,0,0,0.5);
  border-radius: 50%;
  outline: none;
}

.c9:hover,
.c9:focus {
  cursor: pointer;
  box-shadow: 0 0 1px 3px rgba(103,58,183,.5);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 15px;
}

.c0 {
  width: 100%;
}

.c2 .rec-swipable {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  font-size: 0.75rem;
}

.c2 .rec-swipable img {
  max-width: calc(375px - 10rem);
}

.c2 .rec-pagination {
  margin: 0.125rem 0;
}

.c2 .rec-pagination .rec-dot {
  height: 0.75rem;
  width: 0.75rem;
  background: #bdbdbd;
  box-shadow: none;
}

.c2 .rec-pagination .rec-dot_active {
  background: #757575;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c2:last-child .rec-pagination {
  display: none;
}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:768px) {
  .c2 .rec-swipable img {
    max-width: calc(768px - 20rem);
  }
}

@media (min-width:1024px) {

}

<div
  className="c0"
>
  <div
    className="c1 rec rec-carousel-wrapper c2"
    style={Object {}}
  >
    <div
      className="c3 rec rec-carousel"
      size={
        Object {
          "height": 0,
        }
      }
      style={
        Object {
          "height": 0,
        }
      }
    >
      <div
        className="c4 rec rec-slider-container"
      >
        <div
          className="c5 rec rec-slider"
          style={
            Object {
              "left": "0px",
              "right": "auto",
              "top": "auto",
              "transition": "all 500ms ease",
            }
          }
        >
          <div
            className="rec rec-swipable"
            onMouseDown={[Function]}
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              className="rec rec-carousel-item rec-carousel-item-0 rec-carousel-item-visible"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={0}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <img
                  alt="carousel-slide-1"
                  src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_one.svg"
                  tabIndex={0}
                />
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-1 rec-carousel-item-hidden rec-carousel-item-next"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={1}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <img
                  alt="carousel-slide-2"
                  src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_two.svg"
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-2 rec-carousel-item-hidden"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={2}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <img
                  alt="carousel-slide-3"
                  src="https://assets-juanjosemayorga-website.s3.amazonaws.com/tracker-project-s3/slider_three.svg"
                  tabIndex={-1}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c7 rec rec-pagination"
    >
      <button
        className="c8 rec rec-dot rec rec-dot_active"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      />
      <button
        className="c9 rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
      <button
        className="c9 rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
    </div>
  </div>
  <div
    className="c1 rec rec-carousel-wrapper c2"
    style={Object {}}
  >
    <div
      className="c3 rec rec-carousel"
      size={
        Object {
          "height": 0,
        }
      }
      style={
        Object {
          "height": 0,
        }
      }
    >
      <div
        className="c4 rec rec-slider-container"
      >
        <div
          className="c5 rec rec-slider"
          style={
            Object {
              "left": "0px",
              "right": "auto",
              "top": "auto",
              "transition": "all 500ms ease",
            }
          }
        >
          <div
            className="rec rec-swipable"
            onMouseDown={[Function]}
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              className="rec rec-carousel-item rec-carousel-item-0 rec-carousel-item-visible"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={0}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <p
                  tabIndex={0}
                >
                  Sólo necesitas introducir el ID de tu tablero y nosotros haremos el resto
                </p>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-1 rec-carousel-item-hidden rec-carousel-item-next"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={1}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <p
                  tabIndex={-1}
                >
                  Sabrás el número de miembros con los que cuenta tu tablero, y el nombre de cada uno de ellos
                </p>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-2 rec-carousel-item-hidden"
            >
              <div
                className="c6 rec rec-item-wrapper"
                id={2}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <p
                  tabIndex={-1}
                >
                  Sabrás de una manera rápida el número total de tarjetas que tienes en cada lista
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c7 rec rec-pagination"
    >
      <button
        className="c8 rec rec-dot rec rec-dot_active"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      />
      <button
        className="c9 rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
      <button
        className="c9 rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
    </div>
  </div>
</div>
`;
